### PR TITLE
Fix inconsistent doc of `CMS_verify` and `PKCS7_verify` etc.

### PR DESCRIPTION
**Pull request by DDvO**
* Fix many inconsistencies in documentation of `CMS_verify()` and `PKC7_verify()`
* Add some missing detail in related doc: `CMS_add0_cert.pod`, `PKCS7_sign.pod`, and `PKCS7_sign_add_signer.pod`
* Add missing doc for `PKCS7_add_certificate()` and `PKCS7_add_crl()`
* Change `B<p` to `I<` in `CMS_verify()`, `PKC7_verify()` , `PKCS7_sign.pod`, and `PKCS7_sign_add_signer.pod`
* Minor related code cleanup in `crypto/cms` and `crypto/pkcs7` and minor related cleanup in `pkcs7.h.in`
   
BTW, I wonder why the tremendous overlap between CMS and PKCS7 was introduced.
It should be clear that they are basically the same.
Major consolidation should be done here, as now requested in #18919.

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
This cannot be merged to stable branches because code cleanups are not allowed there. The missingcrypto111.txt should not be touched. If you want the doc fixes on stable branches, please split the PR.

**Comment by DDvO** [comment]
> So I will remove the changes to `missingcrypto111.txt` from this PR and move the code/header refactoring to a separate commit within this PR.

Done, and rebased the PR.
So the only content change here is not to touch `missingcrypto111.txt` anymore.
And marked this PR to pertain to `master` only.

**Comment by t8m** [comment]
> Likely I'll set up a PR that fixes the doc for 1.1.1 - there `missingcrypto111.txt` should be updated, right?
> I'm not sure it's worth having a 3rd PR for fixing the doc in 3.0 - what do you think?

There is no `missingcrypto*.txt` in 1.1.1. However you can do just one PR for 3.0 and handle the inevitable conflict on the missing missingcrypto when cherry-picking it to 1.1.1. As that is a trivial conflict.

File: doc/man3/CMS_verify.pod [hunk 1]
```diff
@@ -21,15 +21,19 @@ CMS_get0_signers - verify a CMS SignedData structure
...
-I<flags> is an optional set of flags, which can be used to modify the verify
-operation.
+The optional I<certs> parameter refers to a set of certificates
+in which to search for signing certificates.
+I<cms> may contain extra untrusted CA certificates that may be used for
+chain building as well as CRLs that may be used for certificate validation.
+I<store> may be NULL or point to
+the trusted certificate store to use for chain verification.
+I<detached_data> refers to the signed data if the content is detached from I<cms>.
+Otherwise I<detached_ata> should be NULL and the signed data must be in I<cms>.
```
Comment (t8m) [inline doc/man3/CMS_verify.pod:34]: Typo: detached_data

File: doc/man3/PKCS7_sign.pod [hunk 2]
```diff
@@ -82,20 +82,24 @@ BIO_new_PKCS7().
...
+In versions of OpenSSL before 1.0.0 the I<signcert> and I<pkey> parameters must
+not be NULL.
 
-PKCS7_sign() is similar to PKCS7_sign_ex() but uses default values of
+PKCS7_sign() is like PKCS7_sign_ex() except that it uses default values of
 NULL for the library context I<libctx> and the property query I<propq>.
+This is retained for API backward compatibiliy.
+
+ int PKCS7_add_certificate(PKCS7 *p7, X509 *x509);
+ int PKCS7_add_crl(PKCS7 *p7, X509_CRL *crl);
```
Comment (t8m) [inline doc/man3/PKCS7_sign.pod:102]: Why this addition?

File: doc/man3/PKCS7_sign_add_signer.pod [hunk 3]
```diff
@@ -66,20 +67,32 @@ If present the SMIMECapabilities attribute indicates support for the following
...
 
-
-PKCS7_sign_add_signers() returns an internal pointer to the PKCS7_SIGNER_INFO
-structure just added, this can be used to set additional attributes
+PKCS7_sign_add_signers() returns an internal pointer to the B<PKCS7_SIGNER_INFO>
+structure just added, which can be used to set additional attributes
 before it is finalized.
 
+PKCS7_add_certificate() adds to the B<PKCS7> structure I<p7> the certificate
+I<cert>, which may be an end-entiry (signer) certificate
```
Comment (t8m) [inline doc/man3/PKCS7_sign_add_signer.pod:75]: Typo: end-entity

File: doc/man3/PKCS7_verify.pod [hunk 4]
```diff
@@ -15,106 +15,117 @@ PKCS7_verify, PKCS7_get0_signers - verify a PKCS#7 signedData structure
...
+the I<certs> parameter (if it is not NULL). Then they are looked up in any
+certificates contained in the I<p7> structure unless B<PKCS7_NOINTERN> is set.
+If any signer's certificates cannot be located the operation fails.
 
 Each signer's certificate is chain verified using the B<smimesign> purpose and
-the supplied trusted certificate store. Any internal certificates in the message
-are used as untrusted CAs. If any chain verify fails an error code is returned.
+using the trusted certificate store I<store> if supplied.
+Any internal certificates in the message, which may have been added using
+L<PKCS7_add_certificate(3)>, are used as untrusted CAs unless B<PKCS7_NOCHAIN>.
```
Comment (t8m) [inline doc/man3/PKCS7_verify.pod:56]: Maybe add: `... is set.`

File: include/openssl/pkcs7.h.in [hunk 5]
```diff
@@ -114,8 +114,8 @@ typedef struct pkcs7_enveloped_st {
 typedef struct pkcs7_signedandenveloped_st {
     ASN1_INTEGER *version;      /* version 1 */
     STACK_OF(X509_ALGOR) *md_algs; /* md used */
-    STACK_OF(X509) *cert;       /* [ 0 ] */
-    STACK_OF(X509_CRL) *crl;    /* [ 1 ] */
+    STACK_OF(X509) *cert;       /* [ 0 ] */ /* name should be 'certificates' */
+    STACK_OF(X509_CRL) *crl;    /* [ 1 ] */ /* name should be 'crl' */
```
Comment (t8m) [inline include/openssl/pkcs7.h.in:118]: crls

**Review by t8m — CHANGES_REQUESTED** [review]

File: doc/man3/CMS_verify.pod (hunk 1 above)
Comment (DDvO) [reply doc/man3/CMS_verify.pod]: Fixed, as well as the other typos.

File: doc/man3/PKCS7_sign.pod (hunk 2 above)
Comment (DDvO) [reply doc/man3/PKCS7_sign.pod]: Oh, this must have been a paste error - removed.
These decls belong to `PKCS7_sign_add_signer.pod` (where I put them as well, in the right place).

File: doc/man3/PKCS7_verify.pod (hunk 4 above)
Comment (DDvO) [reply doc/man3/PKCS7_verify.pod]: Done

**Review by t8m — APPROVED** [review]
@paulidale still OK?

**Review by paulidale — APPROVED** [review]

**Comment by DDvO** [comment]
Merged to `master` - thanks @paulidale and @t8m

**Pull request closed without merging**